keras_nlp/layers/rotary_embedding.py

```diff
@@ -16,9 +16,47 @@
 
 
 class RotaryEmbedding(keras.layers.Layer):
-    def __init__(self, rotary_percentage, max_wavelength=10000):
+    """Rotary positional encoding layer.
+
+        Tbjs layer encodes absolute positional information with rotation matrix and naturally
+        incorporates explicit relative position dependency in self-attention formulation.
+        It layer calculates the rotary encoding with a mix of sine and cosine
+        functions with geometrically increasing wavelengths. Defined and formulized
+        in [RoFormer: Enhanced Transformer with Rotary Position Embedding
+    ](https://arxiv.org/abs/2104.09864v4).
+
+        Takes as input the query and key tensors. The input must have shape
+        [batch_size, num_heads, sequence_length, query_length]. This layer will return
+        new query and key tensors after applying rotational encoding.
+
+        Args:
+            percentage: float. The percentage of attn_head_size over which rotation
+                should be applied. Defaults to 0
+            max_wavelength: int. The maximum angular wavelength of the sine/cosine
+                curves, as described in Attention is All You Need. Defaults to
+                `10000`.
+
+        Examples:
+        ```python
+        rotary_percentage = 0.25
+        batch_size = 16
+        num_heads = 8
+        sequence_length = 256
+        query_length = key_length = 256
+        query = tf.ones((batch_size, num_heads, sequence_length, query_length))
```

Comment:
This does not match our general ordering for dims I think? After projecting to multi-headed space, I believe our shapes will look like `(batch_size, sequence_length, num_heads, head_dim)`. Important to follow KerasNLP conventions here, not the ones we picked up from gpt-neox

Also query_length is a bit of an odd term here, should that be head_dim? Or if this is the token length of the query, how is this different than sequence_length.